fontspec-doc-featset.tex

```diff
@@ -608,6 +608,66 @@ \subsection{Letter spacing}
 In particular, small amounts of letter spacing can be very useful, when setting small caps or all caps titles.
 Also see the OpenType \opt{Uppercase} option of the \feat{Letters} feature (\vref{sec:letters}).
 
+
+\section{Variable fonts} \label{sec:variable}
+
+OpenType variable fonts and Multiple Master fonts are parameterised over
+orthogonal font axes, allowing continuous selection along such features
+as weight, width, and optical size.
+
+Currently OpenType variable fonts are only supported in \LuaTeX, while
+Multiple Master fonts only work with \XeTeX.
+
+\subsection{Optical font sizes} \label{sec:aat-opticalsize}
+Whereas traditional OpenType fonts will have only a few separate
+optical sizes, a Variable or Multiple Master font's optical size can be
+specified over a continuous range. Unfortunately, this flexibility makes
+it harder to create an automatic interface through \LaTeX, and the
+optical size for a Variable or Multiple Master font must always be specified
+explicitly.
+\begin{Verbatim}
+  \fontspec{Minion MM Roman}[OpticalSize=11]
+   MM optical size test                    \\
+  \fontspec{Minion MM Roman}[OpticalSize=47]
+   MM optical size test                    \\
+  \fontspec{Minion MM Roman}[OpticalSize=71]
+   MM optical size test                    \\
+\end{Verbatim}
```

Comment:
Left this in since I don't see why it would be MM specific, but I don't see why this is necessarily the case. Wouldn't it make more sense to not specify the optical size and instead implicitly use the selected font size as optical size for variable fonts?

Reply:
It probably makes since to support both, automatic axis setting based on font size and the ability to override it. It probably also makes since to have options for all registered axis https://docs.microsoft.com/en-us/typography/opentype/spec/dvaraxisreg.

Reply:
@khaledhosny That's the current situation, at least as far as the HarfBuzz renderer is involved. When no design size is specified the current font sized is used for the axis.

Regarding options for all registered axes: Currently the `slnt` and `ital` axes are not directly exposed since I'm not sure about the best user interface for them. E.g. `slnt` is specified to be a counter-clockwise angle in degrees of the slant, but having a `Slant` feature specifying a counter clockwise angle in degrees while the existing `FakeSlant` feature specifies the tangent of the clockwise angle seems extremely confusing. Even more so since quite some actual fonts don't actually implement this as specified and instead often also use a clockwise angle...

Also `ital`, an axis which according the the spec is "not expected to be common" in variable fonts would use with `Italic` a very prominent name which might be better used for more common functionality. Beside that similar compliance issues: While `ital`'s values "[v]alues must be in the range 0 to 1" according to the spec, actual fonts (if they have this axis at all) seem to have highly varying ranges.

Since they are not generally recommended to be used together and some fonts seem to make rather surprising decisions which to use there's also the question if they should completely separate interfaces or if the differences which be hidden behind some high-level flag.

In any case, it seemed that this is something which is better handled in a separate PR.

Reply:
I would support slnt without normalizing the angles, it is counterintuitive but that is what the spec does and it would allow supporting our of spec fonts. I agree that ital axis is less important, it is there actually mainly for the STAT table, but slnt seems important enough to expose and not delegate to the raw option.

Reply:
Added with the name `Slant`.